parsers/ES.py

```diff
@@ -341,28 +355,20 @@ def fetch_exchange(
     ses = session or Session()
 
     data = fetch_and_preprocess_data(
-        EXCHANGE_FUNCTION_MAP[sorted_zone_keys],
+        EXCHANGE_MAPPING[sorted_zone_keys]["zone_ref"],
         ses,
         logger,
         target_datetime,
     )
 
     exchangeList = ExchangeList(logger)
     for event in data:
-        exchanges = {}
+        net_flow = 0.0
```

Comment:
This is not good, this will mean the value will always be 0 even if it's missing.

I would recommend saving the exports in one exchange mapping and imports in another and then merging them.

something like in this psudo code:
```py
exchanges = []
for exchange_key in EXCHANGE_MAPPING[sorted_zone_keys]["codes"]:
  exchange_list = ExchangeList(logger)
  for key in event:
    exchange_list.add_event()

return ExchangeList.merge_exchanges(exchanges, logger).toList()
```
(this can probably be made a bit more efficient)

This will ensure it's not defaulting to zero, that we only record the event if both the imports and exports are known, and re-use existing tested logic.